service/interpreter/persistence.go

```diff
@@ -60,22 +60,16 @@ func RebuildPersistenceManager(provider WorkflowProvider,
 }
 
 func (am *PersistenceManager) GetDataObjectsByKey(request service.GetDataObjectsQueryRequest) service.GetDataObjectsQueryResponse {
-	all := false
+	var dataAttributes []iwfidl.KeyValue
+
 	if len(request.Keys) == 0 {
-		all = true
-	}
-	var res []iwfidl.KeyValue
-	keyMap := map[string]bool{}
-	for _, k := range request.Keys {
-		keyMap[k] = true
-	}
-	for key, value := range am.dataObjects {
-		if keyMap[key] || all {
-			res = append(res, value)
-		}
+		dataAttributes = am.GetAllDataObjects()
+	} else {
+		dataAttributes = LoadDataAttributes(request.Keys, am.GetAllDataObjects())
```

Comment:
I think it's a bit wastful to call `am.GetAllDataObjects()` before `LoadDataAttributes`. Because `GetAllDataObjects` will allocate a new slice/array to hold all the data attributes. Maybe `LoadDataAttributes` can just access to all the DAs directly